### ISSUE TITLE
[Bug]: Scheduler state is silently discarded on a sharded resume

### ISSUE THREAD

**Issue by mantreshkhurana** [Spyxpo/framerai#233]
### Preflight checklist

- [x] I searched existing issues and did not find a duplicate.
- [x] I am running the latest version of the affected branch.

### Affected component

Model (Python)

### Description

The sharded checkpoint saves the scheduler and cannot load it back.

On save, `model/training/checkpoint.py:88` puts `scheduler.state_dict()` into the state as a plain
dictionary of scalars and hands it to `dcp.save`.

On load, `model/training/checkpoint.py:138` seeds the slot with `state["scheduler"] = {}` before
calling `dcp.load`. Distributed checkpointing loads in place, filling the objects already present
in the state dictionary. An empty dictionary has no keys to fill, so nothing is restored.

The guard that follows then hides it. `model/training/checkpoint.py:147` reads
`if scheduler is not None and "scheduler" in state and state["scheduler"]:`, and because the
dictionary is still empty the branch never runs. No error, no warning.

A distributed resume therefore restarts the learning-rate schedule from step zero, replaying the
full warmup, while the optimizer moment estimates are restored correctly. That combination is
worse than either failure alone: a fully warmed optimizer is driven by a learning rate ramping
from zero, and the loss curve shows a visible discontinuity at every resume that is easy to read
as ordinary noise.

### Steps to reproduce

1. Train with sharded checkpointing under `torchrun --nproc_per_node=2` past the warmup, to a
   step where the schedule is well into its cosine decay.
2. Record the learning rate reported in the step log.
3. Stop the run and resume from the sharded checkpoint.
4. The step counter and the optimizer state resume correctly.
5. The reported learning rate restarts from the beginning of the warmup instead of continuing the
   decay, and no message says the scheduler was not restored.

### Expected behavior

A resume should restore the scheduler along with the model and optimizer, so the learning rate
continues from where the run stopped.

The scheduler should be wrapped in a `torch.distributed.checkpoint.stateful.Stateful` adapter so
distributed checkpointing can load into it, or its state should be written into the
`checkpoint_meta.json` sidecar and restored explicitly.

A resume that finds no scheduler state while a scheduler was supplied should fail loudly rather
than continue with a reset schedule.

### Logs and error output

```shell
# before stopping
Step 40000/100000 | loss 2.1043 | lr 8.31e-05 | 3.2 it/s

# after resuming from the sharded checkpoint
Step 40001/100000 | loss 2.1050 | lr 5.00e-08 | 3.2 it/s
Step 40100/100000 | loss 2.4788 | lr 5.02e-06 | 3.2 it/s
```

### Environment

OS: Ubuntu 22.04
Python: 3.12.4
PyTorch: 2.8.0
Node: 22.11.0
GPU/VRAM: 8x A100 80GB
Affected branch: stable

**Comment by kunaldeo001** [comment]
I'd like to take up this issue next.
